### PR TITLE
feat(accounts): deliver web walking skeleton

### LINKED ISSUE #1542: feat(accounts): deliver the production-shaped web walking skeleton
## Parent

- [Wayfinder: Modern Accounts client architecture and migration specification](https://github.com/motech-development/platform/issues/1524)

## What to build

Create the smallest useful `@accounts/web` foundation: the Accounts user can sign in, reach a protected company route, record one real Confirmed sale, attach a PDF, and see the authoritative result in an accessible hosted PWA.

Later capability tickets extend this code in place.

## Acceptance criteria

- [ ] Create one static React 19, TanStack Start, Vite, Breeze UI v3, Apollo Client 4, TanStack Form, and Zod application, hosted on AWS Amplify and delivered as a normal repository Deployment Unit without changing legacy production traffic.
- [ ] Use the existing Amplify GitHub App connection without retaining its one-time bootstrap access token. Supply build-only credentials such as `SENTRY_AUTH_TOKEN` as app-level Amplify Hosting configuration inherited by every branch. Branch deployment automation must not read, set, copy, update, or clear that shared credential; configure it manually when the shared app is created or replaced. Do not create an Accounts-specific Secrets Manager stack or expose these values through client-prefixed environment variables or frontend artifacts.
- [ ] Consume Breeze UI through its public `workspace:*` dependency and use directory-based routes for only the real behaviour implemented by this ticket.
- [ ] Support Auth0 login, protected-route readiness, deep-link restoration, intentional invalid/not-found routes, and complete authenticated-state cleanup on logout or identity change. During an identity transition, keep protected content hidden until Apollo state owned by the previous identity has been cleared.
- [ ] Use the existing AppSync operations, Accounts Owner ID, memory-only Apollo cache, generated types, one company-scoped Transaction subscription, and authoritative refetch after reconnect.
- [ ] Record one Confirmed sale using established validation, UTC Transaction Date, VAT calculation and override, duplicate-submit prevention, retry-preserved input, and authoritative mutation results.
- [ ] Attach a PDF through the existing presigned contract using ordinary instrumented browser fetch with the exact content type and no Auth0 credentials on the signed request. The contract's 30-second presigned URL may remain in ordinary network telemetry; failure and retry preserve the form input, and non-cancellation failures reach ordinary error monitoring.
- [ ] Provide an accessible, responsive shell and clear loading, empty, validation, error, retry, and success states for this journey.
- [ ] Provide an installable online-first PWA with direct-link fallback and a basic public shell when offline. Do not persist authenticated data or queue accounting writes.
- [ ] Add focused tests for the new architecture boundaries and run the relevant existing Chromium Confirmed-sale journey against the hosted replacement.
- [ ] Integrate with normal pull-request, Preview Environment, Release, Develop, and production delivery conventions. Do not add analytics, performance benchmarks, capability/evidence matrices, fault injection, or a separate readiness system.
- [ ] Integrate ordinary hosted error monitoring through the TanStack Start Sentry SDK with private source maps. Preserve original exception messages, causes, stack traces, breadcrumbs, routes, identity/entity context, and relevant operation data. Do not add blanket `beforeSend` or breadcrumb scrubbing. Apply only the targeted durable-credential, raw-file, and unrelated-state exclusions defined by the parent.
- [ ] Do not add placeholder feature routes, a runtime server, Amplify client libraries, a global client store, persisted Apollo entities or drafts, or queued mutation replay.

## Blocked by

- [feat(breeze-ui): add Accounts shell and user-menu lifecycle contracts](https://github.com/motech-development/platform/issues/1541)

### PR DESCRIPTION
**Pull request by mogusbi-motech**
## What changed

- delivers the independent `@accounts/web` React application across the real Auth0, AppSync, subscription, notification, and presigned-storage boundaries
- adds the confirmed-sale journey, PDF upload/viewing, responsive Breeze shell, strict routes, offline shell, guarded PWA updates, recovery states, and privacy-safe release identity
- wires Preview and Develop deployment/evidence generation while retaining the legacy client and inherited confirmed-sale assertion
- records source-equivalent Preview and rebased Develop SHAs, requiring identical Git trees and a successful Accounts web Preview deployment
- adds application-level unit, component, accessibility, fault-recovery, keyboard, and hosted Playwright evidence without separately unit-testing Git or the AWS CLI

## Why

This establishes the production-shaped Accounts replacement walking skeleton requested by the architecture decisions, proving the smallest real accounting journey against the existing production service contracts without routing traffic away from the legacy client.

## Validation

- `yarn workspace @accounts/web typecheck`
- `yarn workspace @accounts/web lint`
- `yarn workspace @accounts/web test --run` — 35 tests passed
- `yarn workspace @accounts/web validate:capabilities` — 26 rows validated
- `yarn workspace @accounts/web playwright test --list` — 12 hosted tests discovered
- `yarn workspace @accounts/web build`
- delivery workflow generation check and 83 generator tests

Hosted Auth0/AppSync/storage/PWA evidence runs in the Preview and Develop workflows.

Closes #1542

## Summary by CodeRabbit

* **New Features**
  * Introduced the Accounts web experience with Auth0-gated protected routes, company selection, dashboards, a transactions ledger, recording and viewing transaction details, attachment PDF download/preview, and live notifications.
  * Added offline-first PWA behavior, including update prompting, stale-chunk recovery, and connectivity status.
  * Expanded hosted preview deployment to reconcile, run Playwright E2E, and validate capability evidence.
* **Bug Fixes**
  * Improved access-token renewal synchronization, offline/online handling, and transaction caching/pagination deduplication.
  * Hardened presigned attachment upload/download flows and added accessibility/error recovery coverage.
* **Tests**
  * Added extensive unit, integration, and hosted E2E + capability-evidence validation for journeys and PWA recovery.

### CODE REVIEW

File: .github/workflows/deploy-to-develop.yml [hunk 1]
```diff
@@ -1596,3 +1596,279 @@ jobs:
...
+      deployments: write
+      contents: write
+      id-token: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@9c091bb21b7c1c1d1991bb908d89e4e9dddfe3e0 # v7.0.0
+        with:
+          ref: ${{ fromJSON(inputs.release-plan).desiredTags['@accounts/web'] }}
+          fetch-depth: 0
```
[the remark this answers is not in the document]
Comment (mogusbi-motech) [reply .github/workflows/deploy-to-develop.yml:1668]: Keeping this intentional exception. The Accounts web deployment script pushes the exact release to the managed hosting branch, and teardown deletes that remote branch. Those operations require `contents: write` and the checkout credentials; changing these jobs to `contents: read` with `persist-credentials: false` would break deployment and teardown. The write permission remains scoped to the Accounts web jobs that perform those pushes.

File: applications/accounts/web/src/data/loaders.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  });
+
+  if (!result.data) {
+    throw new Error('The Transaction did not return data');
+  }
+
+  if (result.data.getTransaction.companyId !== companyId) {
+    notFound({ throw: true });
+  }
+}
```
[the remark this answers is not in the document]
Comment (mogusbi-motech) [reply applications/accounts/web/src/data/loaders.ts:130]: No change here because the API contract is non-null: `getTransaction(id: ID!): Transaction!`, and generated `AccountsWebTransactionQuery` likewise types `getTransaction` as non-null. A missing transaction is returned as a GraphQL error rather than nullable data. The optional access in the component handles the query lifecycle, not a nullable schema field.

File: applications/accounts/web/src/styles.css [hunk 3]
```diff
@@ -0,0 +1,4 @@
+@import '../../../../packages/breeze-ui/src/styles/reset.css';
+@import '../../../../packages/breeze-ui/src/styles/styles.css';
+
+@source './**/*.{ts,tsx}';
```
[the remark this answers is not in the document]
Comment (mogusbi-motech) [reply applications/accounts/web/src/styles.css:6]: No change here. This repository does not configure or run Stylelint for this workspace; the application lint contract is ESLint and it passes. `@source` is a supported Tailwind directive, so adding repository configuration solely for an external CodeRabbit Stylelint invocation would introduce an unused toolchain exception.

File: applications/id/tenant/tenant.yaml [hunk 4]
```diff
@@ -79,6 +79,47 @@ clients:
...
+      infinite_idle_token_lifetime: false
+      infinite_token_lifetime: false
+      leeway: 60
+      rotation_type: rotating
+      token_lifetime: 2592000
+    sso_disabled: false
+    token_endpoint_auth_method: none
+    web_origins:
+      - https://accounts-develop.motechdevelopment.co.uk
+      - https://*.##ACCOUNTS_WEB_APP_ID##.amplifyapp.com
```
[the remark this answers is not in the document]
Comment (mogusbi-motech) [reply applications/id/tenant/tenant.yaml]: Keeping this wildcard for the Accounts Develop client. Preview origins are dynamic `pr-N.<app-id>.amplifyapp.com` branches and cannot be statically enumerated before each PR environment exists. The app-id domain and branch lifecycle are controlled by the AWS account and CI role, previews are deleted during teardown, and the production Auth0 client does not contain this wildcard. Replacing it would require a separate per-PR Auth0 registration and cleanup lifecycle, which is intentionally outside this walking-skeleton change.

File: .github/workflows/teardown-environment.yml [hunk 5]
```diff
@@ -962,3 +963,113 @@ jobs:
...
+
+    permissions:
+      contents: write
+      id-token: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@9c091bb21b7c1c1d1991bb908d89e4e9dddfe3e0 # v7.0.0
+        with:
+          fetch-depth: 0
```
[the remark this answers is not in the document]
Comment (mogusbi-motech) [reply .github/workflows/teardown-environment.yml:988]: Keeping this intentional exception. `applications/accounts/web/infrastructure/teardown.ts` deletes the managed remote hosting branch with `git push origin --delete`, so this job requires `contents: write` and persisted checkout credentials. Changing it to read-only credentials would break preview teardown. This same finding was previously raised on this PR and withdrawn after that requirement was clarified.

File: applications/accounts/web/src/data/presigned-transfer.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+    assertSuccessful(response, operation);
+    return response;
+  } catch (cause: unknown) {
+    if (!isCancellation(cause, signal)) {
+      capturePresignedTransferFailure(cause, operation);
+    }
+
+    throw cause;
+  }
```
[the remark this answers is not in the document]
Comment (mogusbi-motech) [reply applications/accounts/web/src/data/presigned-transfer.ts:33]: No change here because this is explicitly permitted by the ticket contract. #1542 allows the complete 30-second presigned URL, including its signing query parameters, to remain in ordinary Sentry network telemetry and states that no special uninstrumented transport or URL scrubbing is required. Adding a presigned-fetch breadcrumb filter would therefore contradict the specified observability behaviour.

File: applications/accounts/web/src/observability.ts [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+
+    if (url.username) url.username = filteredValue;
+    if (url.password) url.password = filteredValue;
+
+    [...url.searchParams.keys()].forEach((name) => {
+      const normalisedName = normaliseFieldName(name);
+      if (oauthQueryParameters.has(normalisedName)) {
+        url.searchParams.set(name, filteredValue);
+      }
+    });
```
[the remark this answers is not in the document]
Comment (mogusbi-motech) [reply applications/accounts/web/src/observability.ts:64]: No change here because #1542 explicitly permits complete 30-second presigned URLs, including signing query parameters, in ordinary Sentry network telemetry and says they do not require URL scrubbing. `secretFieldNames` still protects matching structured fields; extending `redactUrl` to signed-storage query parameters would remove telemetry that the ticket deliberately allows.

File: .github/delivery/templates/deploy-to-production.yml [hunk 8]
```diff
@@ -384,6 +384,44 @@ jobs:
...
+      - setup
+      - deploy-accounts-api
+
+    env:
+      VITE_APPSYNC_URL: ${{ needs.accounts-api.outputs.appsync-url }}
+      VITE_AUTH0_AUDIENCE: ${{ secrets.ACCOUNTS_APP_AUDIENCE }}
+      VITE_AUTH0_CLIENT_ID: ${{ secrets.ACCOUNTS_APP_CLIENT_ID_PROD }}
+      VITE_AUTH0_DOMAIN: ${{ secrets.AUTH0_DOMAIN_PROD }}
+      VITE_AWS_REGION: ${{ needs.accounts-api.outputs.aws-region }}
+      VITE_SENTRY_DSN: ${{ secrets.SENTRY_DSN_ACCOUNTS_WEB }}
```
[the remark this answers is not in the document]
Comment (mogusbi-motech) [reply .github/delivery/templates/deploy-to-production.yml:403]: No change here because this file is a generator template, not the runnable workflow. During generation, the deploy-accounts-api template job is renamed to the catalog job ID accounts-api and its needs block is rewritten before workflow references are validated. The expression therefore intentionally uses needs.accounts-api, matching the generated job and declared dependency in .github/workflows/deploy-to-production.yml. Changing it to needs.deploy-accounts-api as suggested would leave the generated workflow referencing a job that does not exist. The delivery generator tests and generated-workflow drift check confirm the current ID mapping.

File: applications/accounts/web/infrastructure/deploy.ts [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+          current.jobId,
+          '--output',
+          'json',
+        ]),
+      ) as { job: { summary: AmplifyJob } }
+    ).job.summary;
+  }
+
+  return current;
+}
```
[the remark this answers is not in the document]
Comment (mogusbi-motech) [reply applications/accounts/web/infrastructure/deploy.ts:101]: No deadline added. The implementation now sleeps between Amplify get-job calls and streams build logs, which addresses the tight-polling issue. A maximum-attempt or elapsed-time deadline would require an exact value that is not defined by #1542, the AWS contract, or project policy; inventing one would violate the AGENTS.md No unauthoritative limits rule. The loop terminates when Amplify reports a terminal job state, while the enclosing GitHub Actions workflow supplies the platform execution boundary.

File: .github/workflows/deploy-to-environment.yml [hunk 10]
```diff
@@ -1721,6 +1721,211 @@ jobs:
...
+              "repos/${GITHUB_REPOSITORY}/deployments/${{ steps.deployment.outputs.id }}/statuses" \
+              --input -
+
+      - name: Deploy hosted web
+        id: hosted-deployment
+        shell: bash
+        run: |
+          deployment="$(yarn workspace @accounts/web deploy --stage "$STAGE")"
+          echo "deployment-url=$(jq -r .url <<< "$deployment")" \
+            >> "$GITHUB_OUTPUT"
```
[the remark this answers is not in the document]
Comment (mogusbi-motech) [reply .github/workflows/deploy-to-environment.yml]: Has this been dealt with?

File: .github/workflows/release.yml [hunk 11]
```diff
@@ -132,3 +137,41 @@ jobs:
...
+
+    permissions:
+      contents: read
+      id-token: write
+
+    steps:
+      - name: Checkout Accounts web Release
+        uses: actions/checkout@9c091bb21b7c1c1d1991bb908d89e4e9dddfe3e0 # v7.0.0
+        with:
+          ref: ${{ fromJSON(needs.release.outputs.release-plan).desiredTags['@accounts/web'] }}
```
[the remark this answers is not in the document]
Comment (mogusbi-motech) [reply .github/workflows/release.yml:166]: Has this been fixed and if so, can it be resolved?
